pkg/neorpc/errors.go

```diff
@@ -27,36 +27,211 @@ const (
+	// Transaction is already pooled, but not yet accepted into a block.
+	ErrTxAlreadyPooledCode = -503
+	// ErrInsufficientNetworkFeeCode is returned on RPC request processing error: Validation failed.
+	// Transaction has incorrect (too small per Policy setting) network fee value.
+	ErrInsufficientNetworkFeeCode = -504
+	// ErrPolicyFailCode is returned on RPC request processing error: One of the Policy filters failed.
+	// 	Denied by the Policy contract (one of signers is blocked).
+	ErrPolicyFailCode = -505
+	// ErrInvalidScriptCode is returned on RPC request processing error: Invalid script.
+	// Transaction contains incorrect executable script.
+	ErrInvalidScriptCode = -506
+	// ErrTransactionAttrCode is returned on RPC request processing error: Invalid transaction attribute.
+	// Transaction contains an invalid attribute.
+	ErrTransactionAttrCode = -507
+	// ErrInvalidSignatureCode is returned on RPC request processing error: Invalid signature.
+	//	One of the verification scripts failed.
+	ErrInvalidSignatureCode = -508
+	// ErrInvalidSizeCode is returned on RPC request processing error: Invalid inventory size.
+	// Transaction or its script is too big.
+	ErrInvalidSizeCode = -509
+	// ErrExpiredTransactionCode is returned on RPC request processing error: Expired transaction.
+	// Transaction's ValidUntilBlock value is already in the past.
+	ErrExpiredTransactionCode = -510
+	// ErrInsufficientFundsCode is returned on RPC request processing error: Insufficient funds.
+	// Sender doesn't have enough GAS to pay for all currently pooled transactions.
+	ErrInsufficientFundsCode = -511
+	// ErrInvalidVerificationFunctionCode is returned on RPC request processing error: Invalid verification function.
+	// Contract doesn't have a verify method or this method doesn't return proper value.
+	ErrInvalidVerificationFunctionCode = -512
+)
+
+// Errors related to node configuration and various services.
+const (
+	// ErrSessionsDisabledCode is returned on RPC request processing error: Sessions disabled.
+	// Session support is not enabled on the server.
+	ErrSessionsDisabledCode = -601
+	// ErrOracleNotRunCode is returned on RPC request processing error: Oracle service is not running.
+	// Service is not enabled in the configuration.
+	ErrOracleNotRunCode = -602
+	// ErrOracleFinishedCode is returned on RPC request processing error: Oracle request has already been finished. It uses only in C# node.
+	// 	The oracle request submitted is already completely processed.
+	ErrOracleFinishedCode = -603
+	// ErrOracleRequestNotFoundCode is returned on RPC request processing error: Oracle request is not found.
+	// 	The oracle request submitted is not known to this node.
+	ErrOracleRequestNotFoundCode = -604
+	// ErrOracleNotDesignatedNodeCode is returned on RPC request processing error: Not a designated oracle node.
+	// Oracle service is enabled, but this node is not designated to provide this functionality.
+	ErrOracleNotDesignatedNodeCode = -605
```

Comment:
I doubt you need an additional method (after all these are returned on response submission, so it's just a matter of returning something appropriate from the oracle service). But otherwise it'd be a nice addition to debug things, suppose you're a node submitting a response to NeoGo. You can get some generic error and you can get this one, the latter would be much easier to debug.

Reply:
https://github.com/nspcc-dev/neo-go/issues/3068.